drivers/serial/uart_stm32.c

```diff
@@ -92,13 +108,45 @@ static inline void uart_stm32_set_baudrate(const struct device *dev,
 		return;
 	}
 
-
 #if HAS_LPUART_1
 	if (IS_LPUART_INSTANCE(UartInstance)) {
+		uint32_t lpuartdiv;
+#ifdef USART_PRESC_PRESCALER
+		uint8_t presc_idx;
+
+		for (presc_idx = 0; presc_idx < ARRAY_SIZE(USART_PRESCALER_TAB); presc_idx++) {
+			lpuartdiv = LPUARTDIV_CALC(clock_rate, presc_idx, baud_rate);
+			if (lpuartdiv >= LPUART_BRR_MIN_VALUE && lpuartdiv <= LPUART_BRR_MASK) {
+				break;
+			}
+		}
+
+		if (presc_idx == ARRAY_SIZE(USART_PRESCALER_TAB)) {
+			/*
+			 * Calculate the maximum possible lpuartdiv value, if that is still
+			 * lower than the LPUART_BRR_MIN_VALUE, that means this is the closest
+			 * we can get
+			 */
+			lpuartdiv = LPUARTDIV_CALC(clock_rate, 0, baud_rate);
+			if (lpuartdiv < LPUART_BRR_MIN_VALUE) {
+				presc_idx = 0;
+			} else {
+				presc_idx = ARRAY_SIZE(USART_PRESCALER_TAB) - 1;
+			}
+			LOG_ERR("Unable to set %s to %d", dev->name, baud_rate);
```

Comment:
If there is no way to find a prescaler that allows to match the expected baud rate, user needs to review his application settings. Computing an approx baudrate feels useless to me.

Reply:
> Computing an approx baudrate feels useless to me.

I agree